gridstatus/pjm.py

```
@@ -538,6 +603,447 @@ def _get_key(self):
+        checkboxes,
+        to_check_idx,
+        verbose=False,
+    ):
+        # toggle checkbox
+        params = {
+            "javax.faces.partial.ajax": "true",
+            "javax.faces.source": f"chart1FrmLmpSelection:tblBusAggregates:{to_check_idx}:{form_source_id}",  # noqa: E501
+            "javax.faces.partial.execute": f"chart1FrmLmpSelection:tblBusAggregates:{to_check_idx}:{form_source_id}",  # noqa: E501
+            "javax.faces.partial.render": "globalMessages",
+            "javax.faces.behavior.event": "change",
+            "javax.faces.partial.event": "change",
+            "chart1FrmLmpSelection": "chart1FrmLmpSelection",
+            "chart1FrmLmpSelection:tblBusAggregates:name:filter": "",
+            "chart1FrmLmpSelection:tblBusAggregates:voltage:filter": "",
+            "chart1FrmLmpSelection:tblBusAggregates:station:filter": "",
+            "chart1FrmLmpSelection:tblBusAggregates_scrollState": "0,0",
+        }
+        checkboxes[to_check_idx] = True
+        for idx, is_checked in checkboxes.items():
+            if is_checked:
+                params[
+                    f"chart1FrmLmpSelection:tblBusAggregates"
+                    f":{idx}:{form_source_id}_input"  # noqa: E501
+                ] = "on"
+        response = self._dv_lmp_fetch(
+            dv_session,
+            params,
+            verbose=verbose,
+        )
+        if verbose:
+            print(f"response (truncated) = {response.content[0:256]}", file=sys.stderr)
+
+    def _get_lmp_via_dv(
+        self,
+        date,
+        market: str,
+        end=None,
+        locations="hubs",
+        location_type=None,
+        verbose=False,
+    ):
+        """Get latest LMP data from Data Viewer, which includes RT & DA"""
+        with requests.session() as session:
+            initial_fetch = session.get(DATAVIEWER_LMP_URL)
+
+            dv_session = self._new_dv_session(session, verbose=verbose)
+            if dv_session is None:
```

i would raise this error inside `_new_dv_session`. I dont see advantage to doing it here. by moving it, it keeps this function more readable. same with `chart_ids is None:` and `_dv_lmp_extract_chart_ids` below